workitem/workitem_repository.go

```diff
@@ -110,14 +111,54 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, spaceID uuid.UUID, wi
 	return ConvertWorkItemStorageToModel(wiType, wiStorage)
 }
 
+// LookupIDByNamedSpaceAndNumber returns the work item's ID for the given owner name, space name and item number
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LookupIDByNamedSpaceAndNumber(ctx context.Context, ownerName, spaceName string, wiNumber int) (*uuid.UUID, *uuid.UUID, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "workitem", "LookupIDByNamedSpaceAndNumber"}, time.Now())
+	log.Debug(nil, map[string]interface{}{
+		"wi_number":  wiNumber,
+		"space_name": spaceName,
+		"owner_name": ownerName,
+	}, "Loading work item")
+	query := fmt.Sprintf("select wi.id, wi.space_id from %[1]s wi "+
```

Comment:
```sql
select wi.id from work_items wi 
join spaces s on wi.space_id = s.id 
join identities i on s.owner_id = i.id 
where i.username = 'aslak' 
and wi.number = 199
and lower(s.name) = lower('Home') 
and s.deleted_at is null
```

```
 Nested Loop  (cost=0.56..24.64 rows=1 width=16) (actual time=0.033..0.043 rows=1 loops=1)
   ->  Nested Loop  (cost=0.29..16.33 rows=1 width=16) (actual time=0.023..0.028 rows=1 loops=1)
         ->  Index Scan using idx_identities_username on identities i  (cost=0.14..8.16 rows=1 width=16) (actual time=0.011..0.012 rows=1 loops=1)
               Index Cond: (username = 'aslak'::text)
         ->  Index Scan using spaces_name_idx on spaces s  (cost=0.14..8.16 rows=1 width=32) (actual time=0.006..0.008 rows=1 loops=1)
               Index Cond: ((owner_id = i.id) AND (lower(name) = 'home'::text))
   ->  Index Scan using uix_work_items_spaceid_number on work_items wi  (cost=0.27..8.29 rows=1 width=32) (actual time=0.006..0.007 rows=1 loops=1)
         Index Cond: ((space_id = s.id) AND (number = 199))
 Planning time: 0.437 ms
 Execution time: 0.078 ms
```

requires a new lower(username) index on identities to get it fully index only, but, not a huge difference (beyond the need to read the row from disk) since we're only talking about a filter check of 1 row due to the other index matches.